chore: update changelog for version 2.12.1, 2.12.2, 2.14.0, and 2.14.1 (#10839)
DataDog/dd-trace-py: CHANGELOG.md

Changelogs for versions not listed here can be found at https://github.com/DataDog/dd-trace-py/releases

---

## 2.14.0

### Deprecation Notes
- Tracing

>   - Deprecates the `DD_TRACE_SPAN_AGGREGATOR_RLOCK` environment variable. It will be removed in 3.0.0.

Author: ```suggestion
  - Deprecates the `DD_TRACE_SPAN_AGGREGATOR_RLOCK` environment variable. It will be removed in v3.0.0.
```

>   - Deprecates support for [APM Legacy App Analytics](https://docs.datadoghq.com/tracing/legacy_app_analytics/). This feature and its associated configuration options are deprecated and will be removed in a future release.

Author: ```suggestion
  - Deprecates support for [APM Legacy App Analytics](https://docs.datadoghq.com/tracing/legacy_app_analytics/). This feature and its associated configuration options are deprecated and will be removed in v3.0.0.
```

  - google_generativeai: Introduces tracing support for Google Gemini API `generate_content` calls.  
  See [the docs](https://ddtrace.readthedocs.io/en/stable/integrations.html#google_generativeai) for more information.
  - openai: The OpenAI integration now includes a new `openai.request.client` tag with the possible values `OpenAI/AzureOpenAI` to help differentiate whether the request was made to Azure OpenAI or OpenAI.
  - openai: The OpenAI integration now captures token data from streamed completions and chat completions, if provided in the streamed response. To ensure accurate token data in the traced streamed operation, ensure that the `stream_options={"include_usage": True}` option is set on the completion or chat completion call.

- Profiling
  - Captures `asyncio.Lock` usages with `with` context managers.

- Other
  - botocore: Adds span pointers to some successful AWS botocore spans. Currently only supports S3 PutObject.
  - pymongo: Adds support for pymongo>=4.9.0

> 
> 
> 

Author: ```suggestion

```

---

## 2.14.1

### New Features

  - Code Security (IAST): Always report a telemetry log error when an IAST propagation error raises, regardless of whether the `_DD_IAST_DEBUG` environment variable is enabled or not.

### Bug Fixes

  - tracing: Removes a reference cycle that caused unnecessary garbage collection for top-level spans.
  - Code Security: fix potential memory leak on IAST exception handling.

>   - Fixes endpoint profiling when using libdatadog exporter, either with `DD_PROFILING_EXPORT_LIBDD_ENABLED` or `DD_PROFILING_TIMELINE_ENABLED`.

Reviewer: ```suggestion
  - profiling: Fixes endpoint profiling when using libdatadog exporter, either with `DD_PROFILING_EXPORT_LIBDD_ENABLED` or `DD_PROFILING_TIMELINE_ENABLED`.
```
